### PR TITLE
WS-2609:  Migration to a newer parent pom.

### PR DESCRIPTION
**Pull request by seth-mg**
Changes In Source
- Make `ErrorResponse` `transient` in `CommonRosetteApiException`
- Change visibility of `address` in `UnfieldedAddress` from `public` to _package-private_
- `HttpRosetteApi` add suppression for `"java:HttpClient_must_be_closed"` in `initClient()`
- `JacksonMixinProcessor` add suppression for `"java:S3516"` in `process()`

Change In Test
- Migrate to JUnit 5 and Mockserver 5.14.
- Add some mock data.

Tech Debt
- Checkstyle updates
- PMD updates
- Sonar updates
- Version upgrades
- CI Jenkinsfile that pushes metrics to Sonar

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
     <parent>
         <artifactId>open-source-parent</artifactId>
         <groupId>com.basistech</groupId>
-        <version>2.0.3</version>
+        <version>7.0.1</version>
```
Comment (Katsuya-Tomioka) [inline pom.xml:25]: 7.0.1? (ta 63.0.1) Why?

Comment (seth-mg) [reply pom.xml:25]: I can't go any higher until the new ADM release.  The OSGI test fails.

Comment (Katsuya-Tomioka) [reply pom.xml:25]: Waiting for new ADM to go higher. Thanks Seth!

**Review by Katsuya-Tomioka — APPROVED** [review]
Looks good to me!

**Pull request merged**